### ISSUE TITLE
Build log not showing in `production`

### ISSUE THREAD

**Issue by tpmai22** [Seneca-CDOT/telescope#2951]
**What happened**:
Build log not showing on `production` 
![image](https://user-images.githubusercontent.com/43376325/154322896-6b3d8190-72b2-4aa3-8569-142136d88aa3.png)

**What should have happened**:

**How to reproduce it (as precise as possible)**:

**Anything else we need to know?**:
https://telescope.cdot.systems/deploy/status return `null` on all builds 
**Environment**:

- OS:
- Browser:

**Comment by humphd** [comment]
@manekenpix might know if we're running the latest autodployment server in prod, we might not be.

**Comment by manekenpix** [comment]
I think `production` is running the latest version of the `autodeployment` server. The issue is that, at least once a week, there's a power outage in our "server room" and, when the machines come back and the `autodeployment` server is relaunched, it has no build record to serve, and I think that's why the dashboard displays the error message.

**Comment by humphd** [comment]
"Server room" :)

A perfect opportunity to use Supabase and the [Storage API](https://supabase.com/docs/guides/storage)!  We can push build logs and GitHub data about the build into a Supabase storage bucket, and key the name on the git sha.  So maybe for 

```js
// Upload Log
const { data, error } = await supabase.storage
  .from('builds')
  .upload(`public/${sha}/log`, log);

// Upload Info
const { data, error } = await supabase.storage
  .from('builds')
  .upload(`public/${sha}/json`, buildData);
```

Now we can [access these with public URLs](https://supabase.com/docs/guides/storage#accessing-objects).

I'd love to see someone other than @DukeManh or me tackle this one, and we can mentor/review.  How about it, @joelazwar, @rclee91, @TueeNguyen, @sirinoks, @menghif (since you were on the Supabase call today)?

There is another part to this: using these logs in the Dashboard front-end.  We might also want to create a table that keeps track of every build we did: git sha, result (did it build?), links to objects in buckets.

**Comment by sirinoks** [comment]
O_O what even is `sha`

**Comment by TueeNguyen** [comment]
I'm happy to take this.

**Comment by TueeNguyen** [comment]
@sirinoks, it's the id of the commit, it is hashed using SHA (Secure Hash Algorithm). We usually call it sha

**Comment by DukeManh** [comment]
I totally agree, Supabase needs to be maintained by more people. Also, the best way to understand Supabase is to get your hand dirty and do something with it.

@humphd, why not use the database but the storage API?

**Comment by humphd** [comment]
They are just structureless blobs of data, they don't really belong in the db imho.  This is perfect for object storage.

**Comment by TueeNguyen** [comment]
I guess we're going to upload `.txt` file to storage and read it?

**Comment by humphd** [comment]
Log files are `text/plain` and the data from GitHub is `application/json`.  I wouldn't add `.txt`, though.

**Comment by DukeManh** [comment]
If we create a JS Blob object to upload, its type would still be 'application/json', we could store JSON directly in Postgres for simplicity.

**Comment by humphd** [comment]
This is always true.  You can technically put anything in postgres. But the value of blob storage is that for arbitrary data you aren't indexing, it's just a blob you can request later by URL.

Managing more tables, doing backups, etc. seems like overkill for logs.

**Comment by TueeNguyen** [comment]
Gotta look into what type of file we're going to upload to the storage
